src/dymaptic.GeoBlazor.Core/Components/Widgets/LegendWidget.cs

```diff
@@ -99,4 +112,47 @@ public class LayerInfo : MapComponent
     [Parameter]
     [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
     public int[]? SublayerIds { get; set; }
-}
\ No newline at end of file
+}
+
+/// <summary>
+/// The widget legend style, sets the display style of the legend widget.
+/// <a target="_blank" href=" https://developers.arcgis.com/javascript/latest/api-reference/esri-widgets-Legend.html#style">
+///         ArcGIS
+///         JS API
+///     </a>
+/// </summary>
+public class LegendStyle
```

Comment:
Our typical pattern would have this inherit from `MapComponent` so that you could declare your widget like

```html
<LegendWidget>
    <LegendStyle Type="LegendStyleType.Card" Layout="LegendStyleLayout.Auto" />
</LegendWidget>
```